### DISCUSSION TITLE
External Links: Option to open in new tab (also as per-Pub default option)

### DISCUSSION THREAD

**Discussion by tosteiner** [knowledgefutures/pubpub#1569] [Ideas]
When inserting external links in a Pub draft, the editor includes an option to follow the link in question and will open it in a new tab. Once you publish a version of this Pub, links are rendered to open directly in the same window, which often leads to losing the text you have been reading.

I realize this may be a contentious issue depending on one's personal preferences, but in my view, links (as part of any Pub's main body) ought to open in a new tab (i.e. `target="_blank"` ).

Hence I wanted to raise the issue here to consider introducing a user selection on two levels:
* individual links: include checkbox to enable `target="_blank"` 
* plus a per-Pub rule in the Pub Settings dialogue to define external link behaviour for a given document

**Comment by gabestein** [comment]
Thanks for bringing this up. There's a bit of a divide on the dev team on this one, so it's good to hear how more folks feel. 

I'd say our current thinking is something like: this is an instance where our value of empowering communities conflicts with our value of protecting users from the types of manipulation common in ad-driven business models (to be clear: I'm not accusing you, or anyone, of wanting to engage in that behavior, but thinking about the potential for abuse).

In this case, we have tended to lean a bit towards the latter, because cmd/ctrl-click to open a link in a new tab is a fairly well known option for readers who want to open links in new tabs supported by all common browsers, but there's not really an easy way to browse in the same tab if the author specifies `target="_blank"`. Thus, defaulting to (or allowing people to default to) new tabs, even though it's somewhat common across the web, feels to us like a bit of an anti-pattern that more malicious folks could use to keep users on site longer, which serves the purpose of the author at the expense of the reader's experience. There are definitely exceptions to the rule, and I can see us building the ability to target new pages on a per-link basis within this framework. But probably not allowing folks to default to it.

Anyway, that's where we are currently. We'd love to get some more opinions on it, though, from both authors and readers.

**Reply by Nicolsports** [reply]
How much does it cost to play squash in NYC?

**Reply by Nicolsports** [reply]
The cost of playing at [squash courts NYC](https://www.playatnicol.com/) varies by location. Some clubs require memberships, while others offer hourly court rentals or day passes.

**Comment by aimeediehl** [comment]
I agree with @tosteiner. We're starting our first community and would like the option to be able to link out to our journal's society website, in certain instances, without a reader needing to know cmd/ctrl-click. (Many of the folks in our target audience, believe it or not, likely don't know this command). Many researchers and students are used to and expect the default "new tab" behavior built in to most sites when they click a link—even Word docs and PDFs open each link in a new tab. 

I understand your concerns about a default new tab behavior for every link, but would really appreciate the have it as an option when needed. For us, it would enhance the readers' experience, rather than detract from it. Thanks for listening!

**Comment by mevemuller** [comment]
We agree as well at Codicille. Our beta readers also asked the feature and did not like having to travel back to the page they were reading, especially when they were reading on a phone. It would be great to have the option to open links into a new tab.

**Comment by gabestein** [comment]
Hey folks, this has been implemented as of today!

**Reply by tosteiner** [reply]
🙏🏽 oh, perfect - thanks so much to the whole team!

**Comment by aimeediehl** [comment]
This is awesome news, Gabe! It comes just in time for our official community launch in June.

Thanks so much for making this change happen,
Aimee

Aimee Diehl (she/her)
Managing Editor of Journals, UVM Press
University of Vermont Libraries
***@***.******@***.***>

From: Gabriel Stein ***@***.***>
Reply-To: pubpub/pubpub ***@***.***>
Date: Wednesday, April 26, 2023 at 12:22 PM
To: pubpub/pubpub ***@***.***>
Cc: Aimee Diehl ***@***.***>, Comment ***@***.***>
Subject: Re: [pubpub/pubpub] External Links: Option to open in new tab (also as per-Pub default option) (Discussion #1569)


Hey folks, this has been implemented as of today!
